src/radar_control.py

```diff
@@ -35,6 +35,49 @@
 TIME_PROFILE = False
 
 
+@dataclass
+class CFSParameters:
+    """
+    Parameters used to track clear frequency search
+    """
+
+    cfs_freq: list = field(default_factory=list)
+    cfs_mags: list = field(default_factory=list)
+    cfs_range: list = field(default_factory=dict)
+    cfs_masks: list = field(default_factory=dict)
+    last_cfs_set_time: int = 0
+    last_cfs_power: list = field(default_factory=dict)
+    set_new_freq: bool = True
+
+    def check_update_freq(self, cfs_spectrum, cfs_slices, threshold):
+        """
+        Checks if any scanned frequencies have power levels that
+        exceed the current power of each cfs slice based on a threshold
+
+        :params cfs_spectrum:   Results of the CFS analysis
+        :type   cfs_spectrum:   dataclass
```

Comment:
It would be useful to know where to find the definition of this dataclass, so someone can look it up to determine what is being done to it